java/semantickernel-api/src/main/java/com/microsoft/semantickernel/plugin/KernelPluginFactory.java

```diff
@@ -89,7 +89,34 @@ public class KernelPluginFactory {
      * @return The new plugin.
      */
     public static KernelPlugin createFromObject(Object target, String pluginName) {
-        List<KernelFunction<?>> methods = Arrays.stream(target.getClass().getMethods())
+        Class<?> clazz = target.getClass();
+        return KernelPluginFactory.createFromObject(clazz, target, pluginName);
+    }
+
+    /**
+     * Creates a plugin that wraps the specified target object. Methods decorated with
+     * {@code {@literal @}DefineSKFunction} will be included in the plugin.
+     *
+     * @param clazz      The class to be wrapped.
+     * @param target     The instance of the class to be wrapped.
+     * @param pluginName Name of the plugin for function collection and prompt templates. If the
+     *                   value is {@code null}, a plugin name is derived from the type of the
+     *                   target.
+     * @return The new plugin.
+     */
+    public static KernelPlugin createFromObject(Class<?> clazz, Object target, String pluginName) {
+        if (!clazz.isInstance(target)) {
+            throw new SKException("Target object is not an instance of the provided class");
+        }
+
+        if (clazz.getName().endsWith("_ClientProxy")) {
```

Comment:
Isn't this dependent on the DI framework? Could this work as, if reflection fails then warn that a probable cause is proxy? Another possible reason would be a module that doesn't export or open.

Reply:
It is framework specific, I will also add a warning in the case of no methods found